core/element_unary_operations.h

```diff
@@ -105,6 +105,79 @@ constexpr auto atan_out_arg = overloaded{
       x = atan(y);
     }};
 
+auto unit_check_and_assign = [](units::Unit &a, const units::Unit &b,
+                                const units::Unit &repl) {
+  expect::equals(b, repl);
+  a = b;
+};
+
+auto unit_check_and_return = [](const units::Unit &x, const units::Unit &repl) {
+  expect::equals(x, repl);
+  return x;
+};
+
+constexpr auto nan_to_num = overloaded{
+    transform_flags::expect_all_or_none_have_variance, [](const auto x,
+                                                          const auto
+                                                              &repl) noexcept {
```

Comment:
You are running into the same clang-format bug(?) I struggled with. What worked for me was to remove the `noexcept`. I think removing it is ok, cause it is inline anyway.